langchain/src/experimental/prompts/handlebars.ts

```diff
@@ -0,0 +1,107 @@
+      // @ts-expect-error - handlebars' hbs.AST.ContentStatement isn't exported
+      const text = node.value;
+      parsed.push({ type: "literal", text });
+    } else if (node.type === "MustacheStatement") {
+      // @ts-expect-error - handlebars' hbs.AST.MustacheStatement isn't exported
+      const name: string = node.path.parts[0];
+      // @ts-expect-error - handlebars' hbs.AST.MustacheStatement isn't exported
+      const { original } = node.path as { original: string };
+      if (
+        !!name &&
+        !original.startsWith("this.") &&
+        !original.startsWith("@")
+      ) {
+        parsed.push({ type: "variable", name });
+      }
+    } else if (node.type === "PathExpression") {
+      // @ts-expect-error - handlebars' hbs.AST.PathExpression isn't exported
+      const name: string = node.parts[0];
+      // @ts-expect-error - handlebars' hbs.AST.PathExpression isn't exported
+      const { original } = node;
+      if (
+        !!name &&
+        !original.startsWith("this.") &&
+        !original.startsWith("@")
+      ) {
+        parsed.push({ type: "variable", name });
+      }
+    } else if (node.type === "BlockStatement") {
+      // @ts-expect-error - handlebars' hbs.AST.BlockStatement isn't exported
+      nodes.push([...node.params, ...node.program.body]);
+    }
+  }
+
+  return parsed;
+};
+
+export const interpolateHandlebars = (
+  template: string,
+  values: InputValues
+) => {
+  const compiled = Handlebars.compile(template, { noEscape: true });
+  return compiled(values);
+};
+
+export type HandlebarsPromptTemplateInput<RunInput extends InputValues> =
+  CustomFormatPromptTemplateInput<RunInput>;
+
+export class HandlebarsPromptTemplate<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  RunInput extends InputValues = any
+> extends CustomFormatPromptTemplate<RunInput> {
+  static lc_name() {
+    return "CustomPromptTemplate";
+  }
+
+  lc_serializable = false;
+
+  constructor(input: CustomFormatPromptTemplateInput<RunInput>) {
+    super(input);
+    Object.assign(this, input);
+
+    if (this.validateTemplate && this.templateValidator !== undefined) {
+      let totalInputVariables: string[] = this.inputVariables;
+      if (this.partialVariables) {
+        totalInputVariables = totalInputVariables.concat(
+          Object.keys(this.partialVariables)
+        );
+      }
+      this.templateValidator(this.template, totalInputVariables);
+    }
+  }
+
+  /**
+   * Load prompt template from a template
+   */
+  static fromTemplate<
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    RunInput extends InputValues = any[]
+  >(
+    template: string,
+    params: Omit<
+      HandlebarsPromptTemplateInput<RunInput>,
+      "template" | "inputVariables" | "customParser" | "templateValidator"
+    >
+  ) {
+    return super.fromTemplate<RunInput>(template, {
+      ...params,
+      customParser: parseHandlebars,
+      renderer: interpolateHandlebars,
```

Comment:
We should set `validateTemplate: false` here. Validating the handlebars templates with the f-string code doesn't make sense and throws errors:

```
Error: Invalid prompt schema: Single '}' in template.
    at checkValidTemplate (/Users/fant/repos/ai/langchainjs/langchain-core/dist/prompts/template.cjs:88:15)
    at new PromptTemplate (/Users/fant/repos/ai/langchainjs/langchain-core/dist/prompts/prompt.cjs:53:50)
    at new CustomFormatPromptTemplate (/Users/fant/repos/ai/langchainjs/langchain/dist/experimental/prompts/custom_format.cjs:10:9)
    at new HandlebarsPromptTemplate (/Users/fant/repos/ai/langchainjs/langchain/dist/experimental/prompts/handlebars.cjs:60:9)
    at Function.fromTemplate (/Users/fant/repos/ai/langchainjs/langchain/dist/experimental/prompts/custom_format.cjs:49:16)
    at Function.fromTemplate (/Users/fant/repos/ai/langchainjs/langchain/dist/experimental/prompts/handlebars.cjs:80:22)
    at Object.<anonymous> (/Users/fant/repos/ai/godmode/packages/hydra/src/runnables/agent/planTool/default/prompt.ts:76:30)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._compile (/Users/fant/repos/ai/godmode/packages/node_modules/source-map-support/source-map-support.js:568:25)
    at Module.m._compile (/private/var/folders/j2/stdvm9c50l109dgfdqv45x400000gn/T/ts-node-dev-hook-25024792164166465.js:69:33)
```